AEPTarget/Sources/TargetState.swift

```diff
@@ -23,9 +23,37 @@ class TargetState {
 
     private(set) var thirdPartyId: String?
     private(set) var tntId: String?
-    private(set) var sessionTimestampInSeconds: Int64?
+    private(set) var sessionTimestampInSeconds: Int64? {
+        didSet {
+            if sessionTimestampInSeconds == 0 {
+                dataStore.remove(key: TargetConstants.DataStoreKeys.SESSION_TIMESTAMP)
+            } else {
+                dataStore.set(key: TargetConstants.DataStoreKeys.SESSION_TIMESTAMP, value: sessionTimestampInSeconds)
+            }
+        }
+    }
 
-    private(set) var storedSessionId: String
+#if DEBUG
```

Comment:
just checking:

Wondering if we can remove the if DEBUG flag and keep setting storedSessionId as still private
Aaand for the Functional Test, we can test setSessionId and GetSessionID together.. so we don't have to mock targetState.sessionId

Reply:
Yup. I've revisited this code since we need to be able to update the sessionId from Target extension class as well. In summary, exposed another method in state that allows updating (and resetting) the session Id.